packages/core/src/main.ts

```diff
@@ -147,13 +148,15 @@ export async function createLibp2pInstance(
   libp2p._dht._lan._network._protocol = HOPR_DHT_LAN_PROTOCOL
   libp2p._dht._lan._topologyListener._protocol = HOPR_DHT_LAN_PROTOCOL
 
-  const onConnection = libp2p.upgrader.onConnection
-
-  // @TODO implement whitelisting support
+  const onConnectionOriginal = libp2p.upgrader.onConnection
+  const onConnection = async (conn: Connection) => {
```

Comment:
Please add this directly as `libp2p.upgrade.onConnection` - no need to call a function which only calls another function. Blows up LoC and decreases readability